core-drawer-panel.html

```diff
@@ -75,344 +75,425 @@
-      if (this.narrow) {
-        this.selected = this.defaultSelected;
-      }
-      this.setAttribute('touch-action', this.swipeAllowed() ? 'pan-y' : '');
-      this.fire('core-responsive-change', {narrow: this.narrow});
-    },
-    
-    forceNarrowChanged: function() {
-      this.queryMatchesChanged();
-    },
-
-    swipeAllowed: function() {
-      return this.narrow && !this.disableSwipe;
-    },
-    
-    isMainSelected: function() {
-      return this.selected === 'main';
-    },
-
-    startEdgePeek: function() {
-      this.width = this.$.drawer.offsetWidth;
-      this.moveDrawer(this.translateXForDeltaX(this.rightDrawer ?
-          -this.edgeSwipeSensitivity : this.edgeSwipeSensitivity));
-      this.peeking = true;
-    },
-
-    stopEdgePeak: function() {
-      if (this.peeking) {
-        this.peeking = false;
-        this.moveDrawer(null);
-      }
-    },
+      closeDrawer: function() {
+        this.selected = 'main';
+      },
 
-    downHandler: function(e) {
-      if (!this.dragging && this.isMainSelected() && this.isEdgeTouch(e)) {
-        this.startEdgePeek();
-      }
-    },
-
-    upHandler: function(e) {
-      this.stopEdgePeak();
-    },
-    
-    tapHandler: function(e) {
-      if (e.target && this.toggleAttribute && 
-          e.target.hasAttribute(this.toggleAttribute)) {
-        this.togglePanel();
-      }
-    },
+      onQueryMatchesChanged: function(event) {
+        this.narrow = event.detail.value || this.forceNarrow;
 
-    isEdgeTouch: function(e) {
-      return !this.disableEdgeSwipe && this.swipeAllowed() &&
-        (this.rightDrawer ?
-          e.pageX >= this.offsetWidth - this.edgeSwipeSensitivity :
-          e.pageX <= this.edgeSwipeSensitivity);
-    },
+        if (this.narrow) {
+          this.selected = this.defaultSelected;
+        }
 
-    trackStart : function(e) {
-      if (this.swipeAllowed()) {
-        this.dragging = true;
+        this.setAttribute('touch-action', this.swipeAllowed() ? 'pan-y' : '');
+        this.fire('core-responsive-change', {narrow: this.narrow});
+      },
 
-        if (this.isMainSelected()) {
-          this.dragging = this.peeking || this.isEdgeTouch(e);
-        }
+      forceNarrowChanged: function() {
+        this.queryMatchesChanged();
+      },
 
-        if (this.dragging) {
-          this.width = this.$.drawer.offsetWidth;
-          this.transition = false;
-          e.preventTap();
-        }
-      }
-    },
-
-    translateXForDeltaX: function(deltaX) {
-      var isMain = this.isMainSelected();
-      if (this.rightDrawer) {
-        return Math.max(0, isMain ? this.width + deltaX : deltaX);
-      } else {
-        return Math.min(0, isMain ? deltaX - this.width : deltaX);
-      }
-    },
+      swipeAllowed: function() {
+        return this.narrow && !this.disableSwipe;
+      },
 
-    trackx : function(e) {
-      if (this.dragging) {
+      isMainSelected: function() {
```

Comment:
We don't have a set convention for preferring accessors or methods but.. uses of this might read a little more nicely if it was an accessor. What are your thoughts?